### PR TITLE
Fixes to the formatting of `README.rst`

### PR DESCRIPTION
**Pull request by richvdh**
Fixes a couple of formatting errors which were introduced in #12475.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -55,7 +55,7 @@ solutions. The hope is for Matrix to act as the building blocks for a new
 generation of fully open and interoperable messaging and VoIP apps for the
 internet.
 
-Synapse is a Matrix "homeserver" implementation developed by the matrix.org core
```
Comment (DMRobertson) [inline README.rst:58]: Thanks. I need to figure out how to configure PyCharm to not do this when it automatically wraps a line.

File: README.rst [hunk 2]
```diff
@@ -294,13 +294,13 @@ directory of your choice::
     cd synapse
 
 Synapse has a number of external dependencies. We maintain a fixed development
-environment using [poetry](https://python-poetry.org/). First, install poetry. We recommend
+environment using `Poetry <https://python-poetry.org/>`_. First, install poetry. We recommend::
```
Comment (DMRobertson) [inline README.rst:297]: Arg. Is there a ReST linter that might have caught this? Maybe https://pypi.org/project/restructuredtext-lint/

Comment (DMRobertson) [reply README.rst:297]: FWIW, my training/tradition is to treat displayed items as a noun in the flow of a sentence, including punctuation if ending the sentence. Though this doesn't work brilliantly for source code.

**Review by DMRobertson — APPROVED** [review]

File: README.rst (hunk 1 above)
Comment (richvdh) [reply README.rst:58]: I don't think this trailing space was your doing, tbh, and it's not exactly the end of the world anyway. But my editor likes to strip them...

File: README.rst (hunk 2 above)
Comment (richvdh) [reply README.rst:297]: > Arg. Is there a ReST linter that might have caught this? Maybe https://pypi.org/project/restructuredtext-lint/

I've never looked into this, tbh. Given the README is the only RST thing we have nowadays, I'm not sure it's worth much linting effort, though I'll not statnd in the way.

Comment (richvdh) [reply README.rst:297]: > FWIW, my training/tradition is to treat displayed items as a noun in the flow of a sentence, including punctuation if ending the sentence. Though this doesn't work brilliantly for source code.

Afraid I'm not really sure what you're referring to here. The capitalisation of "Poetry"? The `::` ?

Comment (richvdh) [reply README.rst:297]: > I've never looked into this, tbh. Given the README is the only RST thing we have nowadays, I'm not sure it's worth much linting effort, though I'll not statnd in the way.

Further, it's not obvious what there is to "catch": it was valid RST, it just didn't do what you wanted it to do. markdown-style links are a common error that we could look for explicitly, I suppose, but I'm not sure it's reasonable to expect any generic RST linter to do so.

Comment (DMRobertson) [reply README.rst:297]: > > FWIW, my training/tradition is to treat displayed items as a noun in the flow of a sentence, including punctuation if ending the sentence. Though this doesn't work brilliantly for source code.
> 
> Afraid I'm not really sure what you're referring to here. The capitalisation of "Poetry"? The `::` ?

Sorry. I meant the omission of the trailing `::`. The intent was that the following is all one sentence:

> We recommend `<block of shell>`, as described here.

But that's a mionr style thing that isn't worth squabbling over.

Comment (DMRobertson) [reply README.rst:297]: I want a program that knows all of the mistakes that I make in advance and catches them. AKA, the moon on a stick.

Comment (richvdh) [reply README.rst:297]: > Sorry. I meant the omission of the trailing `::`. The intent was that the following is all one sentence:

Oh right. `::` in RST has special syntactic meaning (https://docutils.sourceforge.io/docs/user/rst/quickref.html#literal-blocks). Not sure if there is a way to get that effect without also getting a `:` in the output.

**Pull request merged**